handlers/Subscribe.cpp

```diff
@@ -0,0 +1,233 @@
+#include <boost/json.hpp>
+#include <reporting/server/session.h>
+
+static std::unordered_set<std::string> validStreams { 
+    "ledger",
+    "transactions" };
+
+boost::json::value
+validateStreams(boost::json::object const& request)
+{
+    if (!request.at("streams").is_array())
+    {
+        return "missing or invalid streams";
+    }
+
+    boost::json::array const& streams = request.at("streams").as_array();
+
+    for (auto const& stream : streams)
+    {
+        if (!stream.is_string())
+        {
+            return "streams must be strings";
+        }
+
+        std::string s = stream.as_string().c_str();
+
+        if (validStreams.find(s) == validStreams.end())
+        {
+            return boost::json::string("invalid stream " + s);
+        }
+    }
+
+    return nullptr;
+}
+
+void
+subscribeToStreams(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::array const& streams = request.at("streams").as_array();
+
+    for (auto const& stream : streams)
+    {
+        std::string s = stream.as_string().c_str();
+
+        if (s == "ledger")
+            manager.subLedger(session);
+        else if (s == "transactions")
+            manager.subTransactions(session);
+        else
+            assert(false);
+    }
+}
+
+void
+unsubscribeToStreams(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::array const& streams = request.at("streams").as_array();
+
+    for (auto const& stream : streams)
+    {
+        std::string s = stream.as_string().c_str();
+
+        if (s == "ledger")
+            manager.unsubLedger(session);
+        else if (s == "transactions")
+            manager.unsubTransactions(session);
+        else
+            assert(false);
+    }
+}
+
+boost::json::value
+validateAccounts(boost::json::object const& request)
+{
+    if (!request.at("accounts").is_array())
+    {
+        return "accounts must be array";
+    }
+
+    boost::json::array const& accounts = request.at("accounts").as_array();
+
+    for (auto const& account : accounts)
+    {
+        if (!account.is_string())
+        {
+            return "account must be strings";
+        }
+
+        std::string s = account.as_string().c_str();
+        auto id = ripple::parseBase58<ripple::AccountID>(s);
+
+        if (!id)
+        {
+            return boost::json::string("invalid account " + s);
+        }
```

Comment:
Can we add the ability to accept accounts as 128 bit hex blobs too, instead of just base58?

Reply:
Yeah, I'm gonna change all instance of `parseBase58<AccountID>()` to use `accountFromStringStrict()` in RPCHelpers, and then adjust `accountFromStringStrict()` to be able to parse hex strings